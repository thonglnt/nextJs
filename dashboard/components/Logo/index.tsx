import Image from 'next/image'
import Link from 'next/link'
import { memo } from 'react'

// Constants
import { ROUTE } from '@constants/route'

import primaryLogo from '../../public/logo/primaryLogo.svg'

const Logo = () => {
  return (
    <Link
      href={ROUTE.HOME_PAGE}
      style={{
        maxWidth: '40px',
        maxHeight: '40px',
      }}
      prefetch={false}
    >
      <Image
        src={primaryLogo}
        width={40}
        height={40}
        alt="logo"
        quality={40}
        loading="eager"
      />
    </Link>
  )
}

export default memo(Logo)
