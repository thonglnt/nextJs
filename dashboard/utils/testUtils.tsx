import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
