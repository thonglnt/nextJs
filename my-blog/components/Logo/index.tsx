'use client'

import { Typography, TypographyProps, useTheme } from '@mui/material'
import Link from 'next/link'
import { memo, ReactNode } from 'react'

export interface ILogo extends TypographyProps {
  children: ReactNode | string
  link: string
}
const Logo = ({ children = 'Logo', link = '/', ...props }: ILogo) => {
  const theme = useTheme()

  return (
    <Typography
      variant="h1"
      color={theme.palette.common.black}
      fontSize="30px"
      fontWeight={400}
      textTransform="uppercase"
      {...props}
    >
      <Link href={link}>{children}</Link>
    </Typography>
  )
}

export default memo(Logo)
