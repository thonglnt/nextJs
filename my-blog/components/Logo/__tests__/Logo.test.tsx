import { render } from '@utils/testUtils'
import Logo, { ILogo } from '..'

describe('Logo Component', () => {
  const mockProps: ILogo = {
    children: 'beautyoncode',
    link: '#',
  }

  it('Should show match Logo DOM Snapshot', () => {
    const { container } = render(<Logo {...mockProps} />)

    expect(container).toMatchSnapshot()
  })
})
