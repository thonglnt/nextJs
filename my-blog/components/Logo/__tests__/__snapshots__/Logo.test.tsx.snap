// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component Should show match Logo DOM Snapshot 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-1eixk0k-MuiTypography-root"
  >
    <a
      href="#"
    >
      beautyoncode
    </a>
  </h1>
</div>
`;
